### PR TITLE
Replace boost with standard library equivalents

### PR DESCRIPTION
**Pull request by Dr15Jones**

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: DataFormats/Common/interface/ValueMap.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <map>
 #include <iterator>
 #include <algorithm>
+#include <stddef.h>
```
Comment (ianna) [inline DataFormats/Common/interface/ValueMap.h]: @Dr15Jones - just curious why later cstddef is included and not here:
https://github.com/cms-sw/cmssw/pull/19773/files#diff-1588000c7ee3060dfd82d6c7ea0c1b4bR8

Comment (Dr15Jones) [inline DataFormats/Common/interface/ValueMap.h]: When I googled where the missing header came from, my first hit was `stddef.h`. Probably using `cstddef` would have been a better choice.

**Comment by Dr15Jones** [comment]
@smuzaffar the Jenkin's log seem to show the build timed out but our integration system doesn't seem to have noticed.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by smuzaffar** [comment]
@Dr15Jones , I will fix the jenkins jobs to run some post build in case job failed or timed out

**Comment by Dr15Jones** [comment]
Code builds fine this time

**Comment by Dr15Jones** [comment]
@smuzaffar @davidlange6 the build timed out after 2h but looks like everything ran fine up till then. 

The only changes for **db** to sign is the inclusion of a missing header.

**Comment by davidlange6** [comment]
merge

**Pull request merged**